vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -0,0 +1,105 @@
+---
+  - name: Get the name of the worker node
+    shell: |
+       kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+       | select(.spec.consumerRef.name | contains("{{ CLUSTER_NAME }}-workers"))
+       | .metadata.name'
+    register: worker_node_name
+
+  - name: Define WORKER_NODE variable with worker node name
+    set_fact:
+      WORKER_NODE: "{{ worker_node_name.stdout }}"
+
+  - name: Reboot "{{ WORKER_NODE }}"
+    shell: |
+       kubectl annotate bmh "{{ WORKER_NODE }}" -n metal3 reboot.metal3.io=
+
+  - name: Rebooted node status
+    delegate_to: "{{ CLUSTER_APIENDPOINT_IP }}"
+    vars:
+      ansible_user: "{{ IMAGE_USERNAME }}"
+      ansible_ssh_private_key_file: "{{ SSH_PRIVATE_KEY }}"
+      ansible_python_interpreter: /usr/bin/python3
+    become: yes
+    become_user: "{{ IMAGE_USERNAME }}"
+    block:
+      - name: Wait until rebooted k8s node "{{ WORKER_NODE }}" becomes NotReady
+        shell: kubectl get nodes | grep -w NotReady | wc -l
+        environment:
+          KUBECONFIG: "/home/{{ IMAGE_USERNAME }}/.kube/config"
+        retries: 150
+        delay: 3
+        register: not_ready_nodes
+        failed_when: >
+          (not_ready_nodes.stderr != "") or
+          (not_ready_nodes.rc != 0) or
+          (not_ready_nodes.stdout != "1")
+        until: not_ready_nodes.stdout == "1"
+
+      - name: Wait until rebooted k8s node becomes Ready again
+        shell: kubectl get nodes | grep -w Ready | wc -l
+        environment:
+          KUBECONFIG: "/home/{{ IMAGE_USERNAME }}/.kube/config"
+        retries: 150
+        delay: 3
+        register: ready_nodes
+        failed_when: >
+          (ready_nodes.stderr != "") or
+          (ready_nodes.rc != 0) or
+          (ready_nodes.stdout != "2")
+        until: ready_nodes.stdout == "2"
+
+  - name: Power off "{{ WORKER_NODE }}"
+    shell: |
+       kubectl annotate bmh "{{ WORKER_NODE }}" -n "{{ NAMESPACE }}" reboot.metal3.io/poweroff=
+
+  - pause:
+      minutes: 1
+
+  - name: Powered off node status
+    delegate_to: "{{ CLUSTER_APIENDPOINT_IP }}"
+    vars:
+      ansible_user: "{{ IMAGE_USERNAME }}"
+      ansible_ssh_private_key_file: "{{ SSH_PRIVATE_KEY }}"
+      ansible_python_interpreter: /usr/bin/python3
+    become: yes
+    become_user: "{{ IMAGE_USERNAME }}"
+    block:
+      - name: Wait until powered off k8s node "{{ WORKER_NODE }}" becomes NotReady
+        shell: kubectl get nodes | grep -w NotReady | wc -l
+        environment:
+          KUBECONFIG: "/home/{{ IMAGE_USERNAME }}/.kube/config"
+        retries: 150
+        delay: 3
+        register: powered_off_node
+        failed_when: >
+          (powered_off_node.stderr != "") or
+          (powered_off_node.rc != 0) or
+          (powered_off_node.stdout != "1")
+        until: powered_off_node.stdout == "1"
+
```

Comment:
you should add something to verify that the vm is properly off IMO

Reply:
like checking virsh list

Reply:
Virsh checking added